### ISSUE TITLE
expected return values from script?

### ISSUE THREAD

**Issue by mtk64** [zb0th/homebridge-ssh#8]
Hi :-)
could you please provide some more information or some examples of how to use return values/strings from scripts beeing called within homebridge-ssh?

I am doings this, calling a script to read the value:

"accessories": [
              {
              "accessory": "SSH",
              "name": "Steckdose",
              "on": "/aio/gw_send_sc_IT B 3 on",
              "off": "/aio/gw_send_sc_IT B 3 off",
              "state": "/aio/gw_getstate_IT B 3",
              "on_value" : "on",
              "off_value" : "off",
              "exact_match": true,
   ...  
thie called script gw_getstate returns "on" or "off".

e.g.
root@gaja2:~# /aio/gw_getstate_IT B 3
on
root@gaja2:~# 

But this is not reflected on my IOS device. :-(

kind regards
michael

**Comment by alguevara7** [comment]
Hi,

In principle your configuration is correct. Except the off_value line. The accessory is considered to be ON if  the value returned by the "on" script matches the value in "on_value". It is considered to be OFF otherwise.

The value returned by the "on" script is logged to the homebridge console. You should see log entries like:

State of Steckdode is: VALUE

**Comment by cfurgang** [comment]
I agree... If you're looking for an example, here's one that I'm using to start/stop the Plex service running on my NAS.

```
{
        "accessory": "SSH",
        "name": "Plex Media Server",
        "on": "systemctl restart plexmediaserver",
        "off": "systemctl stop plexmediaserver",
        "state": "systemctl status plexmediaserver | grep running | wc -l",
        "on_value": "1",
        "exact_match": false,
        "ssh": {
          "user": "root",
          "host": "plex-vm.local",
          "password": "",
          "key": "/opt/homebridge/homebridge-ssh/rsakey"
        }
    }
```

**Comment by mtk64** [comment]
Many thx to both of you, 
I checked that again and something is still not working when reading the result of a script.

Everythings works fine and I can set any state to my devices via bash script properly.
The correct result comes within the console when using the homekit apps. 
But the result is not reflected within the apps (Eve and MyTouchHome).

I cant believe that both sw provide the same problem. 
When homebrighe-ssh provides the correct result to the console, maybe there is a problem on interface to homebridge?

I also checked out to use 1/0 instead of on/off in the way of  "| grep on | wc -l" and "on_value": "1", to check it against your configuration, but no way ...

Can you see the actual state of  your plexmediaserver on your iphone? 
Maybe a hint for another app I can use to check that again??

**Comment by cfurgang** [comment]
Yes I can see the state correctly of my plex server in the Eve app. I understand that you can't _see_ the state of the switch correctly, but if you set the switch to ON or OFF anyway, does it succeed in sending the "on" command and the "off" command? If that doesn't work either, then something is probably wrong with your SSH configuration...

**Comment by mtk64** [comment]
Yes, the switch is going on and off, so SSH should be ok on this. 
Also the result in the console is reflecting the correct state of the switch.
In Eve there is only a correct result, when I change the state from Eve itself.
If the state is changed by another way, e.g. your plexmediaserver is stopped via CLI,
this comes not to Eve. 
My test setup is: 
two accessories A and B with the same switch behind.
switching A:  

"/aio/gw_send_sc_IT B 3 on" to be executed 
and reading via "/aio/gw_getstate_IT B 3"

Switch B is configured the same, just with another name in config.json
Both switches are working on setting the new state, but the other should go into the same state as well, because it will read the same switch.

**Comment by cfurgang** [comment]
Try setting this simple test accessory up for the same server. Does this work, or do you have the same problem?

```
    {
        "accessory": "SSH",
        "name": "Test switch",
        "on": "echo 1 > /tmp/test_state.txt",
        "off": "echo 0 > /tmp/test_state.txt",
        "state": "cat /tmp/test_state.txt",
        "on_value": "1",
        "exact_match": false,
        "ssh": {
          [...]
        }
    }
```

**Comment by mtk64** [comment]
same problem. can see correct values (on/1 and off/0) in console, in eve both switches are "off" when i enter the room again (there it starts reading and gets no values, allthough I can see them in console ...)

**Comment by cfurgang** [comment]
Try adding this line below line 32 of `homebridge-ssh/index.js` -

 `this.log("Comparing \"%s\" to \"%s\"", this.onValue, match);`

This should make the following function (lines 32-40):

```
SshAccessory.prototype.matchesString = function(match) {
  this.log("Comparing \"%s\" to \"%s\"", this.onValue, match);
  if(this.exactMatch) {
    return (match === this.onValue);
  }
  else {
    return (match.indexOf(this.onValue) > -1);
  }
}
```

Then try the test accessory again and let us know what the output of the console is. Maybe that will enlighten us on exactly what data is being received by SSH...

**Comment by alguevara7** [comment]
Great idea. I will push a PR tonight to improve logging in general.

**Comment by mtk64** [comment]
Hi guys, i just learned a lot! Trying to modify the index.js i realized that it is not the same you are using. This function was missing in my version. So I made some "npm install -g homebridge-ssh" to update. Sorry for that, I just installed my homebridge a few days ago so I did not expect that right now.
After doing that I tested the scenario again and its is working right now. It was some outdated code causing my problem here. 

Some last question: Is it possible to assign scripts to rooms, scenarios a.s.o. immediately from configuration in homebridge-ssh? 

Would like to place my assignments into my configuration management because I had to reset my homekit configuration on ipad for several times, and all these asigments went lost.
My hope is to get this from homebridge ... but I do not know the architecture. if this data is only stored within apples homekit cloud it might be impossible...
